### PR TITLE
Upgrade to .net 5

### PR DESCRIPTION
**Pull request by eladiw**
The following changes are proposed:

* Upgrading from .net core 3 to .net 5

### CODE REVIEW

File: K2Bridge/K2Bridge.csproj [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
     <CodeAnalysisRuleSet>CodeAnalysis.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup>
     <NoWarn>$(NoWarn);NU1701</NoWarn>
+    <NoWarn>$(NoWarn);NU1202</NoWarn>
```
[the remark this answers is not in the document]
Comment (meguittet) [reply K2Bridge/K2Bridge.csproj:10]: @eladiw could you commit this PR into 7.10 branch please (the team will create it). Thank you

File: K2Bridge/Controllers/QueryController.cs [hunk 2]
```diff
@@ -231,7 +232,8 @@ private void CheckEncodingHeader()
             // Since by default we ignore this, adding a log to see if it does happen.
             if (logger.IsEnabled(LogLevel.Warning))
             {
-                var hasEncodingHeader = HttpContext.Request.Headers?.TryGetValue("accept-encoding", out var encodingData);
+                StringValues encodingData = string.Empty;
+                var hasEncodingHeader = HttpContext.Request.Headers?.TryGetValue("accept-encoding", out encodingData);
```
Comment (AsafMah) [inline K2Bridge/Controllers/QueryController.cs:236]: Why this change?

File: K2Bridge/Controllers/QueryController.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ namespace K2Bridge.Controllers
     using Microsoft.AspNetCore.Http;
     using Microsoft.AspNetCore.Mvc;
     using Microsoft.Extensions.Logging;
+    using Microsoft.Extensions.Primitives;
```
Comment (AsafMah) [inline K2Bridge/Controllers/QueryController.cs:21]: Why do we need the extra import?

**Review by tomconte — APPROVED** [review]

File: K2Bridge/Controllers/QueryController.cs (hunk 3 above)
Comment (eladiw) [reply K2Bridge/Controllers/QueryController.cs:21]: because of the new code on line 235

File: K2Bridge/Controllers/QueryController.cs (hunk 2 above)
Comment (eladiw) [reply K2Bridge/Controllers/QueryController.cs:236]: because the compiler now complains this value might not be initialized

**Review by meguittet — APPROVED** [review]

File: K2Bridge/Controllers/QueryController.cs (hunk 2 above)
Comment (AsafMah) [reply K2Bridge/Controllers/QueryController.cs:236]: Ok, got it

**Review by AsafMah — APPROVED** [review]

**Review by dupuyjs — APPROVED** [review]
